Source/WebCore/workers/service/InstallEvent.cpp

```diff
@@ -196,14 +196,22 @@ void InstallEvent::addRoutes(ScriptExecutionContext& context, std::variant<Route
         return;
     }
 
-    // FIXME: We should delay installing the service worker until addRoutes async task is finished.
+    auto& globalObject = *JSC::jsCast<JSDOMGlobalObject*>(promise->globalObject());
```

Comment:
This appears to assume `globalObject` is non-null. I don't see how that assumption can be safe, but perhaps I'm missing something?
Seems like we should store as a pointer and null check it before referencing.

Reply:
Done